pennylane/devices/qubit/jaxpr_adjoint.py

```diff
@@ -0,0 +1,163 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Compute the jvp of a jaxpr using the jacobian jacobian method.
+"""
+import jax
+from jax import numpy as jnp
+from jax.interpreters import ad
+
+from pennylane import adjoint, generator
+from pennylane.capture import disable, enable
+from pennylane.capture.primitives import AbstractOperator
+
+from .apply_operation import apply_operation
+from .initialize_state import create_initial_state
+
+
+def _read(env, var):
+    """Return the value and tangent for a variable."""
+    return (var.val, ad.Zero(var.aval)) if isinstance(var, jax.core.Literal) else env[var]
+
+
+def _operator_forward_pass(eqn, env, ket):
+    """Apply an operator during the forward pass of the adjoint jvp."""
+    invals, tangents = tuple(zip(*(_read(env, var) for var in eqn.invars)))
+    op = eqn.primitive.impl(*invals, **eqn.params)
+    env[eqn.outvars[0]] = (op, ad.Zero(AbstractOperator()))
+
+    if any(not isinstance(t, ad.Zero) for t in tangents[1:]):
+        raise NotImplementedError("adjoint jvp only differentiable parameters in the 0 position.")
```

Comment:
Does this stem from the restriction that the adjoint method only works for single-parameter/single-generator gates? Is the positional restriction an additional restriction? Maybe a dev comment could clarify :)

Reply:
limitation for the moment to handle both the fact that generators are define with respect a single parameter, and it doesn' t make sense for wires to be differentiable.